validate.go

```diff
@@ -4,13 +4,171 @@ import (
 	"fmt"
 	"strings"
 
-	onelog "github.com/francoispqt/onelog"
+	appsv1 "github.com/kubewarden/k8s-objects/api/apps/v1"
+	batchv1 "github.com/kubewarden/k8s-objects/api/batch/v1"
 	corev1 "github.com/kubewarden/k8s-objects/api/core/v1"
+	metav1 "github.com/kubewarden/k8s-objects/apimachinery/pkg/apis/meta/v1"
 	kubewarden "github.com/kubewarden/policy-sdk-go"
+	kubernetes "github.com/kubewarden/policy-sdk-go/pkg/capabilities/kubernetes"
 	kubewarden_protocol "github.com/kubewarden/policy-sdk-go/protocol"
 	"github.com/mailru/easyjson"
 )
 
+const DEPLOYMENT_KIND = "deployment"
+const REPLICASET_KIND = "replicaset"
+const STATEFULSET_KIND = "statefulset"
+const DAEMONSET_KIND = "daemonset"
+const REPLICATIONCONTROLLER_KIND = "replicationcontroller"
+const CRONJOB_KIND = "cronjob"
+const JOB_KIND = "job"
+const POD_KIND = "pod"
+
+func getNamespace(validationRequest kubewarden_protocol.ValidationRequest) (*corev1.Namespace, error) {
+
+	host := getWapcHost()
+
+	resourceRequest := kubernetes.GetResourceRequest{
+		APIVersion: "v1",
+		Kind:       "Namespace",
+		Name:       validationRequest.Request.Namespace,
```

Comment:
Warning: if the user enforced with policy targeting a cluster-wide resource, `validation.Request.Namespace` might be empty or nil (I don't know what is the default for this go struct). Please a check for that and return an error